kie-remote/kie-remote-client/src/test/java/org/kie/services/client/builder/http/HeaderTest.java

```diff
@@ -0,0 +1,216 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.services.client.builder.http;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import org.junit.Test;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.remote.client.api.RemoteRuntimeEngineFactory;
+import org.kie.remote.client.jaxb.ClientJaxbSerializationProvider;
+import org.kie.remote.client.jaxb.JaxbCommandsRequest;
+import org.kie.remote.client.jaxb.JaxbCommandsResponse;
+import org.kie.services.client.serialization.JaxbSerializationProvider;
+import org.kie.services.client.serialization.jaxb.impl.process.JaxbProcessInstanceResponse;
+import org.simpleframework.common.buffer.Allocator;
+import org.simpleframework.common.buffer.FileAllocator;
+import org.simpleframework.http.Request;
+import org.simpleframework.http.Response;
+import org.simpleframework.http.core.Container;
+import org.simpleframework.http.core.ContainerTransportProcessor;
+import org.simpleframework.transport.SocketProcessor;
+import org.simpleframework.transport.TransportProcessor;
+import org.simpleframework.transport.TransportSocketProcessor;
+import org.simpleframework.transport.connect.Connection;
+import org.simpleframework.transport.connect.SocketConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class HeaderTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(RedirectTest.class);
+
+    private static final String DEFAULT_ENPOINT = "/rest/execute";
+
+    private static final String TEST_HEADER_NAME = UUID.randomUUID().toString();
+    private static final String ANOTHER_TEST_HEADER_NAME = UUID.randomUUID().toString();
+    private static final String NOT_SENT_HEADER_NAME = UUID.randomUUID().toString();
+
+    private static final Random random = new Random();
+
+    @Test
+    public void testSetHeader() throws Exception {
+        int port = AvailablePortFinder.getNextAvailable(1025);
+        HttpHeaderServer server = new HttpHeaderServer(port);
+
+        server.start();
```

Comment:
The above code looks to me like it should be in a ```@Before``` method. And, more importantly, there seems to be no ```server.stop()```.